packages/widgets/src/blocks/legacy-widget/edit/preview.js

```diff
@@ -13,34 +13,35 @@ import { Placeholder, Spinner, Disabled } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 
 export default function Preview( { idBase, instance, isVisible } ) {
-	const [ iframeHeight, setIframeHeight ] = useState();
+	const [ isLoaded, setIsLoaded ] = useState( false );
 
 	// Resize the iframe on either the load event, or when the iframe becomes visible.
 	const ref = useRefEffect( ( iframe ) => {
-		function onChange() {
-			const boundingRect = iframe?.contentDocument?.body?.getBoundingClientRect();
-			if ( boundingRect ) {
-				// Include `top` in the height calculation to avoid the bottom
-				// of widget previews being cut-off. Most widgets have a
-				// heading at the top that has top margin, and the `height`
-				// alone doesn't take that margin into account.
-				setIframeHeight( boundingRect.top + boundingRect.height );
-			}
+		function setHeight() {
+			iframe.style.height = `${ iframe.contentDocument.documentElement.offsetHeight }px`;
```

Comment:
As mentioned in the comment, still seeing some scrollbars on the gallery widget. For that widget, the gallery's `body` is taller than its `document`, which is why this calculation makes it too short. Probably why I needed to do something different in the previous PR.

I think it might be worth checking both the document and the body and taking the `max` value. That would hopefully work just using `offsetHeight` instead of `getBoundingClientRect`, but I don't know for sure 😓 .

Reply:
Yeah, that's my first thought too. I'll run some test for the gallery block. Thanks!

Reply:
It seems like the gallery block has negative margins that make `<body>` larger than `<html>`. Even if we set the height of the iframe to the height of `<body>`, there's still going to have scrollbars because `<html>`'s size doesn't change.

I believe we have to fix it in core 🤔.

Reply:
Are you sure? When I checked, `iframe.contentDocument.documentElement.body.offsetHeight` had the correct height for the gallery.

Reply:
Oops I was wrong. The issue remained though.

1. There still are negative margins which stretch the width to be greater than 100%, result in a horizontal scrollbar.
2. The value we get from `offsetHeight` seems to be truncated to an integer, which is not enough for floating point heights.

I guess we can solve the vertical scrollbar issue but might have to do more work on the horizontal scrollbar.

Reply:
I wonder if I can just set overflow to hidden for the preview frame, is it too aggressive?

Reply:
> I wonder if I can just set overflow to hidden for the preview frame, is it too aggressive: 48a5b4e?

I personally think it's ok to modify the document or the body, as long as the content isn't being touched.

It could be wrapped in a try/catch to prevent any cross origin issues if that's a concern.